src/core/public/chrome/nav_links/to_nav_link.ts

```diff
@@ -29,6 +34,28 @@ export function toNavLink(app: PublicAppInfo, basePath: IBasePath): NavLinkWrapp
   });
 }
 
+export function toNavDeepLink(
+  app: PublicAppInfo,
+  deepLink: PublicAppDeepLinkInfo,
+  basePath: IBasePath
+): NavLinkWrapper {
+  const defaultNavStatus = deepLink.navLinkStatus === AppNavLinkStatus.default;
+  const relativeBaseUrl = basePath.prepend(`${app.appRoute!}${deepLink.path!}`);
+  const url = relativeToAbsolute(appendAppPath(relativeBaseUrl, app.defaultPath));
+  const baseUrl = relativeToAbsolute(relativeBaseUrl);
+
+  return new NavLinkWrapper({
+    ...deepLink,
+    ...(app.category ? { category: app.category } : {}),
+    deepLinkPath: relativeBaseUrl,
+    hidden: defaultNavStatus || deepLink.navLinkStatus === AppNavLinkStatus.hidden,
+    disabled: deepLink.navLinkStatus === AppNavLinkStatus.disabled,
+    baseUrl,
+    href: url,
+    url,
```

Comment:
Can we reuse or factorize the code from `toNavLink` here, somehow? That seems like some avoidable duplication for just a few additions to the logic